### ISSUE TITLE
Setup continuous integration and automatic deployment

### ISSUE THREAD

**Issue by ghuntley** [planetxamarin/planetxamarin#5]
_Related Issues:_
- #4

**Comment by ghuntley** [comment]
![image](https://cloud.githubusercontent.com/assets/127353/21968985/91b2fc34-dbee-11e6-8e2a-637717e55c76.png)

CI has been setup, all that remains is to make the Azure PaaS website automatically deploy a new version when changes are merged into master. GitHub has been configured to disallow merges unless CI passes and the pull-request has been reviewed.

**Comment by jamesmontemagno** [comment]
So, I have the site setup, how do I get it so you can deploy to it? Did you create a new team for this and add people on in appveyor?

**Comment by ghuntley** [comment]
AppVeyor does checks to prevent merges from occurring that would cause breakages. Once the unit of work has been merged into GitHub a webhook on this repository is configured to send a signal to Azure to do a new deployment.

You now have permission to manage the organization and configure webhooks on his repository James. Refer to the following for instructions on how to setup the webhook.

- https://github.com/projectkudu/kudu/wiki/Deploying-from-GitHub
- https://github.com/projectkudu/kudu/wiki/Deployment-credentials#site-level-credentials-aka-publish-profile-credentials

**Comment by ghuntley** [comment]
ACL's on AppVeyor have been reconfigured after the move to GitHub organization..

![image](https://cloud.githubusercontent.com/assets/127353/21975298/fdde4db0-dc1f-11e6-861a-08773b3b71e7.png)
![image](https://cloud.githubusercontent.com/assets/127353/21975301/ffe2490e-dc1f-11e6-8d08-cf3630981d38.png)

**Comment by jamesmontemagno** [comment]
Azure now has this built in with a continuous delivery preview, going to try that out... I just need you to confirm the request for github access.

**Comment by ghuntley** [comment]
Request has been approved.

**Comment by jamesmontemagno** [comment]
Nice. Soooo I am thinking we should do:'

Dev == Main branch to send PRs into & merge -> Gets deployed automatically to staging

Once a week we do a merge into master, which puts to production.... thoughts?

**Comment by ghuntley** [comment]
Single branch... master... Don't merge unless you want it deployed. Less admin overhead. Put energy instead into #20

**Comment by jamesmontemagno** [comment]
sounds good, so we have it so no one can push to master correct and everything goes through PRs.

**Comment by ghuntley** [comment]
Yup.

**Comment by jamesmontemagno** [comment]
I believe it is working now... will check

**Comment by jamesmontemagno** [comment]
Good to go!
